### PR TITLE
Add infra build support for ppc64le platform

### PR DESCRIPTION
**Pull request by mkumatag**
This PR Addresses:

- Enables build scripts to build multi arch docker images
- Infra Dockerfiles for ppc64le platform

### CODE REVIEW

**Comment by mkumatag** [comment]
Test results @ https://gist.github.com/mkumatag/48e858098e0dbd953dfc9fd2bfc5444a

File: examples/hello-openshift/Dockerfile.ppc64le [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM scratch
+MAINTAINER Jessica Forrester <jforrest@redhat.com>
```
Comment (stevekuznetsov) [inline examples/hello-openshift/Dockerfile.ppc64le]: Might want to change this to yourself or someone else who's signed on to this

File: hack/lib/build/constants.sh [hunk 2]
```diff
@@ -2,8 +2,16 @@
 
 # This script provides constants for the Golang binary build process
 
-readonly OS_BUILD_ENV_GOLANG="${OS_BUILD_ENV_GOLANG:-1.7}"
-readonly OS_BUILD_ENV_IMAGE="${OS_BUILD_ENV_IMAGE:-openshift/origin-release:golang-${OS_BUILD_ENV_GOLANG}}"
+OS_BUILD_ENV_GOLANG="${OS_BUILD_ENV_GOLANG:-1.7}"
+OS_BUILD_ENV_IMAGE="${OS_BUILD_ENV_IMAGE:-openshift/origin-release:golang-${OS_BUILD_ENV_GOLANG}}"
+
+if [[ "$(go env GOARCH)" == "ppc64le" ]]; then
```
Comment (stevekuznetsov) [inline hack/lib/build/constants.sh]: Not sure this makes sense -- we want this to be a reproducible release image build for the state of the code, that can't really vary by arch

File: hack/build-base-images.sh [hunk 3]
```diff
@@ -5,10 +5,29 @@
...
 tag_prefix="${OS_IMAGE_PREFIX:-"openshift/origin"}"
 
+BASE_EXTRA_TAG=""
+if [[ "${HOST_ARCH}" == "amd64" ]]; then
+  BASE_EXTRA_TAG="${tag_prefix}-base"
+fi
+
 # Build the base image without the default image args
-OS_BUILD_IMAGE_ARGS="${OS_BUILD_IMAGE_BASE_ARGS-}" os::build::image "${OS_ROOT}/images/base" "${tag_prefix}-base"
+OS_BUILD_IMAGE_ARGS="${OS_BUILD_IMAGE_BASE_ARGS-}" os::build::image "${OS_ROOT}/images/base" "${tag_prefix}-base-${HOST_ARCH}" ${DOCKERFILE} ${BASE_EXTRA_TAG}
```
Comment (stevekuznetsov) [inline hack/build-base-images.sh]: We didn't need to do this before (for `centos7` and `rhel7`) by curating the images that are available at build-time on the local daemon. Why can't that approach work for this?

File: examples/hello-openshift/Dockerfile.ppc64le (hunk 1 above)
Comment (mkumatag) [inline examples/hello-openshift/Dockerfile.ppc64le]: Sure, I'll change it my Name..

File: hack/lib/build/constants.sh (hunk 2 above)
Comment (mkumatag) [inline hack/lib/build/constants.sh]: I would love to but we have some issues with 1.7.x release and fixes went into 1.8. Even in k8s we are generating the binaries using golang 1.8.

File: hack/build-base-images.sh (hunk 3 above)
Comment (mkumatag) [inline hack/build-base-images.sh]: While building docker images I'm planning to have <imagename>-<arch> to support multi arch like amd64, ppc64le etc.. to fallback to support existing model I'm tagging still tagging them as <imagename> only for amd64 without arch in it. 

e.g openshift/origin-base and openshift/origin-base-amd64 for amd64 arch
openshift/origin-base-ppc64le for ppc64le platform..

**Comment by mkumatag** [comment]
@smarterclayton can you PTAL at this PR?

File: images/base/Dockerfile.ppc64le [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#
+# This is the base image from which all OpenShift Origin images inherit. Only packages
+# common to all downstream images should be here. Depends on Centos 7.2+.
+#
+# The standard name for this image is openshift/origin-base
+#
+FROM ppc64le/centos:centos7
```
Comment (smarterclayton) [inline images/base/Dockerfile.ppc64le]: We've changed how base images work - you'd have a pp64le "source" image that is different, and everything else would be the same.

Comment (smarterclayton) [inline images/base/Dockerfile.ppc64le]: So you should no longer need *any* image that isn't the same as base.

Comment (mkumatag) [inline images/base/Dockerfile.ppc64le]: @smarterclayton  I agree with that, my biggest concern is - we should have different images for different architectures like `openshift/origin-pod-amd64` for amd and `openshift/origin-pod-ppc64le` or we should create manifest image `openshift/origin-pod` which will point to all architecture images?

**Pull request closed without merging**